pkg/controllers/datastoresyncer/datastoresyncer.go

```diff
@@ -103,6 +111,41 @@ func (d *DatastoreSyncer) Run(stopCh <-chan struct{}) error {
 		return err
 	}
 
+	if len(d.localCluster.Spec.GlobalCIDR) > 0 {
+		nodeName, ok := os.LookupEnv("NODE_NAME")
+		if !ok {
+			// Healthcheck in globalnet deployments will not work because of missing NODE_NAME.
+			klog.Error("Error reading the NODE_NAME from the env, healthChecker functionality will not work.")
+		} else {
+			d.localNodeName = nodeName
+			d.localFederator = syncer.GetLocalFederator()
```

Comment:
We might as well set this above and use it in `createLocalCluster` and `createOrUpdateLocalEndpoint` instead of passing it into those methods.

Reply:
Ack, done.